### PR TITLE
core: Replace logging macros

### PR DESCRIPTION
**Pull request by daniellimws**
This replaces (approximately) half of the logging macros in modules under core. 

Now we are only left with core/hle. I decided to leave it to another PR as there are a lot of lines to be replaced under that module, and there's already a lot in this PR.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/3820)

### CODE REVIEW

**Comment by daniellimws** [comment]
Is this PR too big? Should I close it and open a PR for each module above?

**Comment by Dragios** [comment]
Should be fine. It's just trivial changes.

File: src/core/file_sys/archive_extsavedata.cpp [hunk 1]
```diff
@@ -87,48 +87,48 @@ class ExtSaveDataArchive : public SaveDataArchive {
 
     ResultVal<std::unique_ptr<FileBackend>> OpenFile(const Path& path,
                                                      const Mode& mode) const override {
-        LOG_DEBUG(Service_FS, "called path=%s mode=%01X", path.DebugStr().c_str(), mode.hex);
+        NGLOG_DEBUG(Service_FS, "called path={} mode={:01X}", path.DebugStr(), mode.hex);
```
Comment (mailwl) [inline src/core/file_sys/archive_extsavedata.cpp]: mode=0x{:01X}?

Comment (daniellimws) [reply src/core/file_sys/archive_extsavedata.cpp]: The original version didn't have `0x` so I don't think it should be there.

File: src/core/file_sys/archive_selfncch.cpp [hunk 2]
```diff
@@ -236,19 +236,17 @@ class SelfNCCHArchive final : public ArchiveBackend {
...
     if (app_loader.ReadProgramId(program_id) != Loader::ResultStatus::Success) {
-        LOG_WARNING(
+        NGLOG_WARNING(
             Service_FS,
             "Could not read program id when registering with SelfNCCH, this might be a 3dsx file");
     }
 
-    LOG_DEBUG(Service_FS, "Registering program %016" PRIX64 " with the SelfNCCH archive factory",
-              program_id);
+    NGLOG_DEBUG(Service_FS, "Registering program {} with the SelfNCCH archive factory", program_id);
```
Comment (wwylele) [inline src/core/file_sys/archive_selfncch.cpp]: should be `{:016X}`

File: src/core/file_sys/archive_selfncch.cpp [hunk 3]
```diff
@@ -236,19 +236,17 @@ class SelfNCCHArchive final : public ArchiveBackend {
...
 
     if (ncch_data.find(program_id) != ncch_data.end()) {
-        LOG_WARNING(Service_FS,
-                    "Registering program %016" PRIX64
-                    " with SelfNCCH will override existing mapping",
-                    program_id);
+        NGLOG_WARNING(Service_FS,
+                      "Registering program {} with SelfNCCH will override existing mapping",
```
Comment (wwylele) [inline src/core/file_sys/archive_selfncch.cpp]: should be `{:016X}`

File: src/core/memory.cpp [hunk 4]
```diff
@@ -38,8 +38,8 @@ PageTable* GetCurrentPageTable() {
 }
 
 static void MapPages(PageTable& page_table, u32 base, u32 size, u8* memory, PageType type) {
-    LOG_DEBUG(HW_Memory, "Mapping %p onto %08X-%08X", memory, base * PAGE_SIZE,
-              (base + size) * PAGE_SIZE);
+    NGLOG_DEBUG(HW_Memory, "Mapping {} onto {:08X}-{:08X}", memory, base * PAGE_SIZE,
```
Comment (wwylele) [inline src/core/memory.cpp]: please convert `memory` to `(void*)`

**Comment by Hexagon12** [comment]
Please fix the conflicts.

**Pull request merged**